internal/dcache/rpc/server/handler.go

```diff
@@ -46,44 +56,743 @@ var _ service.ChunkService = &ChunkServiceHandler{}
 
 // ChunkServiceHandler struct implements the ChunkService interface
 type ChunkServiceHandler struct {
-	locks *common.LockMap
-	// more fields can be added here as needed
+	locks   *common.LockMap
+	fsIDMap map[string]*rvInfo // map to store the fsID to rvInfo mapping
+}
+
+type rvInfo struct {
+	rvID     string       // rv0, rv1, etc.
+	cacheDir string       // cache dir path for this RV
+	mvMap    sync.Map     // sync map of MV id against mvInfo
+	mvCount  atomic.Int64 // count of MVs for this RV, this should be updated whenever a MV is added or removed from the sync map
+}
+
+type mvInfo struct {
+	syncInfo                  // sync info for this MV
+	peerRVs      []string     // sorted list of peer RVs for this MV
+	chunkOpsCnt  atomic.Int64 // count of chunk operations (get, put or remove) for this MV in the given RV
+	blockOpsFlag atomic.Bool  // block chunk operations for this MV in the given RV. This is done when the .sync dir contents are being moved to the regular MV dir
+}
+
+type syncInfo struct {
+	mu        sync.Mutex
+	isSyncing bool   // is the MV in syncing state
+	syncID    string // sync ID for this MV if it in syncing state
+
+	// sourceRV is the RV that is syncing the target RV in the MV.
+	// This is used when the source RV goes offline and is replaced by the cluster manager with a new RV.
+	// In this case, the sync will need to be restarted from the new RV.
+	sourceRV string // source RV for syncing this MV
 }
 
 func NewChunkServiceHandler() *ChunkServiceHandler {
+	// TODO: get fsID, rvID and cache dir path for different RVs for the node from cluster manager
+	// below will be call to cluster manager to get the information
+	fsIDMap := make(map[string]*rvInfo)
+
 	return &ChunkServiceHandler{
-		locks: common.NewLockMap(),
+		locks:   common.NewLockMap(),
+		fsIDMap: fsIDMap,
+	}
+}
+
+// check if the given mv is valid
+func (rv *rvInfo) isMvValid(mvPath string) error {
+	if !common.DirectoryExists(mvPath) {
+		return fmt.Errorf("MV path %s does not exist", mvPath)
+	}
+
+	mvID := filepath.Base(mvPath)
+	_, err := rv.getMVInfo(mvID)
+	return err
+}
+
+func (rv *rvInfo) getPeerRVForMV(mvID string) []string {
```

Comment:
getPeerRVsForMV - plural RVs to make it clear that it may return more than one RV

Reply:
Updated